### ISSUE TITLE
Root version of the app does not work with Magisk 26.1

### ISSUE THREAD

**Issue by RishiPrasath** [ThePBone/RootlessJamesDSP#100]
This is the error message:

**Comment by RishiPrasath** [comment]
The original JamesDSP app will crash too. The affected devices are : 
Xperia 1 IV 
Redmi Note 10 Pro
(Both are installed with the latest stable Magisk)

**Comment by coxackie** [comment]
I am not facing this problem, and also using magisk 26.1.

But, beyond this, are you sure this is a problem of the app? Have you tried the original jamesdsp apk? If that is not working as well, it is a problem with the JDSP magisk module, and not this app.

**Comment by RishiPrasath** [comment]
Yeah, it's the problem with the magisk module. I'm using v4.6 of the module.

**Comment by RishiPrasath** [comment]
In fact how did you get it to work?  Would be grateful if can you help,  thanks .

**Comment by coxackie** [comment]
> In fact how did you get it to work? Would be grateful if can you help, thanks .

Nothing really, just updated from 25 to 26.1 and everything continued to work.

**Comment by RishiPrasath** [comment]
Ah, I see; that's what happened. But bad news, if you remove the app and magisk module and reinstall again, you will hit the issue. So I just wanted to let you know.

**Comment by ThePBone** [comment]
The official JDSP app bundled with the Magisk module has no error handling when `libjamesdsp.so` isn't found, so it just crashes. That's the same issue as the error screen posted above.

> However, it does raise the question of whether ThePBone would be interested in creating a full magisk module that installs the library plus his apk

I don't have any clue how these modules are setup/maintained, so I probably won't create one myself. (I also don't know why this issue happens on some devices in the first place; until now I only knew about the 'Huawei' option being a possible workaround).

**Comment by RishiPrasath** [comment]
Except that Huawei doesn't work anymore in Magisk 26

**Comment by Melodycloudu** [comment]
It seems to also apply to A12 LineageOS on the LG v30

**Comment by ThePBone** [comment]
Another person published a patched Magisk module which appears to fix this issue. It also bundles the root version of this app alongside the original JDSP app:

https://github.com/akavkl/JamesDSP-for-Android/releases

**Comment by RishiPrasath** [comment]
I've tried it , but the app still crashes. The affected device is Sony Xperia 1 IV.

**Comment by FailSafeNow** [comment]
I'm having the exact same problem. Got a Moto One 5g Ace. And I miss my James!

**Comment by RishiPrasath** [comment]
Last resort is Magisk Delta. 25.2 based. Only way to get it to work. Latest Jamesdsp 4.7 won't work on Magisk 26.1.

**Comment by a0193143** [comment]
Zackptg5 had released a new version, and it seems to have fixed this issue.
https://github.com/Zackptg5/JamesDSPManager

**Comment by FailSafeNow** [comment]
> Zackptg5 had released a new version, and it seems to have fixed this issue. https://github.com/Zackptg5/JamesDSPManager

I was looking for more information. He didn't mention compatibility with the newer Magisk. How do you know he fixed the issue?
